src/abstractdataframe/where.jl

```diff
@@ -0,0 +1,191 @@
+
+    if df isa AbstractDataFrame
+        df_conditions = select(df, conditions..., copycols=!(df isa DataFrame))
+    else
+        df_conditions = select(df, conditions...,
+                               copycols=!(parent(df) isa DataFrame), keepkeys=false)
+    end
+    test_type = dropmissing ? Union{Missing, Bool} : Bool
+    for col in eachcol(df_conditions)
+        if !(eltype(col) <: test_type)
+            throw(ArgumentError("each transformation must return a vector whose " *
+                                "eltype is subtype of $test_type"))
+        end
+    end
+    @assert ncol(df_conditions) == length(conditions)
+    if ncol(df) == 1
+        return isequal.(df_conditions[!, 1], true)
+    else
+        return isequal.(.&(eachcol(df_conditions)...), true)
+    end
+end
+
+"""
+    where(df::AbstractDataFrame, args...; dropmissing::Bool=true, view::Bool=false)
+    where(gdf::GroupedDataFrame, args...; dropmissing::Bool=true, ungroup::Bool=true)
+
+Return a copy of data frame `df` or parent of `gdf` containing only rows for
+which all values produced by `args` transformation in a given row is `true`. If
+`dropmissing=true` (the default) returing `missing` in `args` is allowed and
+these rows are dropped. If `dropmissing=false` an error is thrown if `missing`
+is present.
+
+Each argument passed in `args` can be either a single column selector or a
+`source_columns => function` transformation specifier following the rules
+described for [`select`](@ref).
+
+Note that as opposed to [`filter`](@ref) the `where` function works on whole columns
+and by default drops rows for which contition is false.
+
+If `view=true` a view instead of a `DataFrame` is returned.
+
+If `ungroup=true` then a `GroupedDataFrame` is returned using the same columns as
+`gdf` to group by.
+
+See also: [`where!`](@ref), [`filter`](@ref), [`filter!`](@ref)
+
+# Examples
+
+```
+julia> df = DataFrame(id=1:4, x=[true, false, true, false], y=[true, true, false, false],
+                      z=[true, true, missing, missing], v=[1, 2, 11, 12])
+4×5 DataFrame
+│ Row │ id    │ x    │ y    │ z       │ v     │
+│     │ Int64 │ Bool │ Bool │ Bool?   │ Int64 │
+├─────┼───────┼──────┼──────┼─────────┼───────┤
+│ 1   │ 1     │ 1    │ 1    │ 1       │ 1     │
+│ 2   │ 2     │ 0    │ 1    │ 1       │ 2     │
+│ 3   │ 3     │ 1    │ 0    │ missing │ 11    │
+│ 4   │ 4     │ 0    │ 0    │ missing │ 12    │
+
+julia> where(df, :x)
+2×5 DataFrame
+│ Row │ id    │ x    │ y    │ z       │ v     │
+│     │ Int64 │ Bool │ Bool │ Bool?   │ Int64 │
+├─────┼───────┼──────┼──────┼─────────┼───────┤
+│ 1   │ 1     │ 1    │ 1    │ 1       │ 1     │
+│ 2   │ 3     │ 1    │ 0    │ missing │ 11    │
+
+julia> where(df, :x, :y => ByRow(!))
+1×5 DataFrame
+│ Row │ id    │ x    │ y    │ z       │ v     │
+│     │ Int64 │ Bool │ Bool │ Bool?   │ Int64 │
+├─────┼───────┼──────┼──────┼─────────┼───────┤
+│ 1   │ 3     │ 1    │ 0    │ missing │ 11    │
+
```

Comment:
Also show a simple example like `:v => (x -> x > 2)`?

Reply:
This will not work. But I will add a working example.